library/std/src/ffi/c_str/tests.rs

```diff
@@ -117,6 +117,43 @@ fn from_bytes_with_nul_interior() {
     assert!(cstr.is_err());
 }
 
+#[test]
+fn cstr_from_bytes() {
+    // Test an empty slice. This should fail because it
+    // does not contain a nul byte.
+    let b = b"";
+    CStr::from_bytes(&b[..]).unwrap_err();
```

Comment:
Instead of unwraps I would expect these to be `assert_eq` macro calls.

Reply:
Thanks @AzureMarker for the feedback.

For the error cases, sure, I can replace `unwrap_err` with `assert_eq!(r, Err(FromBytesError))`. It's a unit type right now, but I can see how it sets a good example (also the error type may change).

If your concern is also about the success cases, I don't see any way to do equality comparisons on a `CStr`. Each of these are equivalent for my purposes:
```
assert_eq!(r.to_str().unwrap(), "hello");   // What I'm currently doing
assert_eq!(r.to_string_lossy(), "hello");   // Creates a Cow, which seems a little awkward
assert_eq!(r.to_bytes(), b"hello");         // Maybe this would be best?
```

Reply:
Yeah, the success cases would have to be a little different. I think your byte string example would work best.

Reply:
`to_bytes` omits the `\0`; there's also `to_bytes_with_nul` :)

Reply:
There's still another `unwrap()` in the success cases, e.g.
```
let b = b"hello\0world!";
let r = CStr::from_bytes(&b[..]).unwrap();
assert_eq!(r.to_bytes(), b"hello");
```

I'm not sure there's a way to remove the unwrap without making the test code a lot less readable. The other tests in this file all do similar things, FWIW.

Reply:
That should be fine.